maturity-level-two/src/Controllers/v1/CustomizationController.cs

```diff
@@ -0,0 +1,186 @@
+        public async Task<IActionResult> DeleteCustomization(int id)
+        {
+            var CustomizationObj = await _coditoRepository.GetCustomizationAsync(id);
+            if (CustomizationObj == null)
+            {
+                return NotFound(new ProblemDetailsError(StatusCodes.Status404NotFound));
+            }
+            await _coditoRepository.DeleteCustomizationAsync(id);
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Update customization
+        /// </summary>
+        /// <param name="id">customization identifier</param>
+        /// <param name="customizationDto">Update for the customization object</param>
+        /// <remarks>Update the customization profile (incremental update)</remarks>
+        /// <returns>Acknowledge the object has been updated</returns>
+        [HttpPatch("{id}", Name = Constants.RouteNames.v1.UpdateCustomizationIncremental)]
+        [SwaggerResponse((int)HttpStatusCode.NoContent, "The data has been updated")]
+        [SwaggerResponse((int)HttpStatusCode.NotFound, "Customization not found")]
+        [SwaggerResponse((int)HttpStatusCode.InternalServerError, "API is not available")]
+        public async Task<IActionResult> UpdateIncremental(int id, [FromBody] CustomizationDto customizationDto)
+        {
+            var CustomizationObj = await _coditoRepository.GetCustomizationAsync(id);
+            if (CustomizationObj == null)
+            {
+                return NotFound(new ProblemDetailsError(StatusCodes.Status404NotFound));
+            }
+
+            var customizationUpdated = Mapper.Map<Customization>(customizationDto);
+            customizationUpdated.Id = id;
+
+            await _coditoRepository.ApplyPatchAsync<Customization, CustomizationDto>(customizationUpdated, customizationDto);
+            return NoContent();
+
+        }
+
+        /// <summary>
+        /// Sell a car customization
+        /// </summary>
+        /// <param name="id">Customization identifier</param>
+        /// <returns>Acknowledge that the c</returns>
+        [HttpPost("{id}/sale", Name = Constants.RouteNames.v1.SellCustomization)]
+        [SwaggerResponse((int)HttpStatusCode.Accepted, "Sale accepted. No response body")]
+        [SwaggerResponse((int)HttpStatusCode.BadRequest, "Out of stock")]
+        [SwaggerResponse((int)HttpStatusCode.NotFound, "Car customization not found")]
+        [SwaggerResponse((int)HttpStatusCode.InternalServerError, "API is not available")]
+        public async Task<IActionResult> VoteAsBestPlayer(int id)
+        {
+            var customization = await _coditoRepository.GetCustomizationAsync(id);
+            if (customization == null)
+            {
+                return NotFound(new ProblemDetailsError(StatusCodes.Status404NotFound));
+            }
+            if (customization.InventoryLevel <= 0)
+            {
+                return BadRequest();
```

Comment:
This should not result in a BadRequest response.

A BadRequest indicates that something is wrong at the client-level; something is wrong with the request object.  This is not the case when you have no inventory left.
A NotAllowed statuscode might be more appropriate although I'm not convinced of it either.

Reply:
@MassimoC We discussed this... NotAllowed is okay for me

Reply:
I created an issue for this. Better to discuss this and make it clear as guideline